### DISCUSSION TITLE
how to create function in Zustand to delete object from array?

### DISCUSSION THREAD

**Discussion by PhilipZi** [neuefische/questions#130] [Web] [answered]
Hallo zusammen, ich würde gerne ein angeklicktes Objekt aus dem Array enfernen, tue mich aber schwer da die richtige funtion zu schreiben kann mir jemand helfen? 
Vielen Dank im Voraus! 

hier das was ich versucht habe: 

```
import create from 'zustand';

const useSavedExercises = create(set => ({
	savedExercises: [],
	addNewExercise: currentExercise => {
		set(state => {
			return {savedExercises: [...state.savedExercises, currentExercise]};
		});
	},
	deleteExercise: currentExercise => {
		set(state => {
			return {savedExercises: [...state.savedExercises, currentExercise.splice()]};
		});
	},
}));

export default useSavedExercises;
````

**Comment by tsimvris** [comment, marked as answer]
Hi Phillip,

That should do the Job 
`deleteExcercise: id => {
				set(state => {
					return {savedExercises: state.excersices.filter(excercise => excersice.id !== id)};
				});
			},
`

**Reply by PhilipZi** [reply]
![Screenshot 2022-08-31 at 15 22 29](https://user-images.githubusercontent.com/102238180/187688544-cd5410e2-d949-46d8-9afb-74c5f2fc812b.png)
er kann das nicht ausführen leider. auch die exercise konnte er nicht finden also (exercise => exercise.id) das war rot

**Reply by alexclaes** [reply]
`services` doesn't exist in your `state`, therefore it is `undefined`.

Make sure to apply `filter()` to the correct data

**Reply by PhilipZi** [reply]
Thanks a lot yes i was confused a bit but now i understand! Thank you worked out! also my issue was that i didnt give any id it works with the name key
